### PR TITLE
Add dydx chain

### PR DESCRIPTION
**Pull request by dylanschultzie**
Still need to figure out what `ibc/8E27BA2D5493AF5636760E354E46004562C46AB7EC0CC4C1CA14E9E20E2545B5` is...

### CODE REVIEW

File: dydx/assetlist.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      ],
+      "base": "adydx",
+      "name": "dYdX",
+      "display": "dydx",
+      "symbol": "DYDX",
+      "logo_URIs": {
+        "png": "https://raw.githubusercontent.com/cosmos/chain-registry/master/dydx/images/dydx.png",
+        "svg": "https://raw.githubusercontent.com/cosmos/chain-registry/master/dydx/images/dydx.svg"
+      },
+      "coingecko_id": "DYDX",
```
Comment (JeremyParish69) [inline dydx/assetlist.json:25]: lowercase 'dydx'

File: dydx/chain.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
+  "$schema": "../chain.schema.json",
+  "chain_name": "dydx",
+  "status": "live",
+  "website": "https://dydx.exchange/",
+  "network_type": "mainnet",
+  "pretty_name": "dYdX Protocol",
+  "chain_id": "dydx-mainnet-1",
+  "bech32_prefix": "dydx",
```
Comment (JeremyParish69) [inline dydx/chain.json:9]: unregistered
https://github.com/satoshilabs/slips/blob/master/slip-0173.md

**Comment by JeremyParish69** [comment]
> Still need to figure out what `ibc/8E27BA2D5493AF5636760E354E46004562C46AB7EC0CC4C1CA14E9E20E2545B5` is...

@dylanschultzie 
8E27... == SHA256('transfer/channel-0/uusdc')

File: dydx/chain.json (hunk 2 above)
Comment (dylanschultzie) [reply dydx/chain.json:9]: https://github.com/satoshilabs/slips/pull/1623

**Comment by dylanschultzie** [comment]
> > Still need to figure out what `ibc/8E27BA2D5493AF5636760E354E46004562C46AB7EC0CC4C1CA14E9E20E2545B5` is...
> 
> @dylanschultzie 8E27... == SHA256('transfer/channel-0/uusdc')

woah how did you figure that out?

**Comment by JeremyParish69** [comment]
> > @dylanschultzie 8E27... == SHA256('transfer/channel-0/uusdc')
> 
> woah how did you figure that out?

oh, I just plugged the '8E27...' hash into the reverseSHA256() function and it gave me the original input.

**Comment by danbryan** [comment]
yeah, can confirm this is the same hash we used for the previous dydx-testnets for bringing noble usdc over.

**Comment by boojamya** [comment]
> oh, I just plugged the '8E27...' hash into the reverseSHA256() function and it gave me the original input.

@JeremyParish69 What is this reverseSHA256() function?

**Comment by JeremyParish69** [comment]
> > oh, I just plugged the '8E27...' hash into the reverseSHA256() function and it gave me the original input.
> 
> @JeremyParish69 What is this reverseSHA256() function?

My apologies for the joke--none such function exists (that I know of) as that would compromise much of modern encryption.

What I really did was I used trial and error to guess the channel, and luckily channel-0 (my first try) ended up yielding the exact hash as we see in the ibc denom.

**Comment by boojamya** [comment]
Ahh yes lol.
Wasn't sure if it was some magical version of `gaiad q ibc-transfer denom-trace`

**Comment by HS-Joe** [comment]
@dylanschultzie it's pretty well explained in https://api-docs.skip.money/docs/ibc-routing-algorithm, worth the read

**Pull request merged**